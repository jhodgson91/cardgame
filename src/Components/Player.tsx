import React, { Component } from 'react';
import './Main.scss'

//Define props types
export interface Props {
  title: string;
  readOnly: boolean;
	turn: boolean;
  theme: string;
  playCard: () => void | undefined;
	snap: () => void | undefined;
}

export default class Player extends React.Component<Props> {

  constructor(props: Props) {
    super(props);
  }

  showButtons() {
    if (!this.props.readOnly && this.props.turn) {
      return (
        <div className="small-2">
          <button className="button" onClick={this.props.playCard}>Play card</button>
					<button className="button" onClick={this.props.snap}>Snap</button>
        </div>
      )
    } else if (!this.props.readOnly && !this.props.turn) {
			return(
				<div className="small-2">
					<button className="button" onClick={this.props.snap}>Snap</button>
        </div>
			)
		}
  }

  render(): React.ReactNode {
    return (
      <div id={this.props.title} className="small-5 player">
        <div className="small-2 small-offset-1">
          <h3>{this.props.title}</h3>
        </div>
        {this.showButtons()}
        <div className={`cell player-cards ${this.props.theme}`} >
          <div className="grid-container">
            <div className="grid-x grid-padding-x">
              {this.props.children}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
