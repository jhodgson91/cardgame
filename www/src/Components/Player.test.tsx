import * as React from 'react';
import * as enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Player from './Player';

it('renders the same as last time with a button', () => {
  //Mock the button clicking
  const mockCallBack = jest.fn();
  const tree = renderer
    .create(
      <Player title="house" readOnly={false} playCard={mockCallBack} theme="?">
        <div id="test">Text</div>
      </Player>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the same as last time without a button', () => {
  //Mock the button clicking
  const mockCallBack = jest.fn();
  const tree = renderer
    .create(
      <Player title="house" readOnly={true} playCard={mockCallBack} theme="?">
        <div id="test">Text</div>
      </Player>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders with a button', () => {
  //Mock the button clicking
  const mockCallBack = jest.fn();
  const player = enzyme.shallow(
    <Player title="house" readOnly={false} playCard={mockCallBack} theme="?">
      <div id="test">Text</div>
    </Player>
  );
  expect(player.find('#house')).toBeDefined;
  expect(player.find('h3').text()).toBe('house');
  expect(player.find('button').text()).toBe('Play card');
  expect(player.find('#test').text()).toBe('Text');
  player.find('button').simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});

it('renders without a button', () => {
  //Mock the button clicking
  const mockCallBack = jest.fn();
  const player = enzyme.shallow(<Player title="house" readOnly={true} theme="?" playCard={() => { }} />);
  expect(player.find('#house')).toBeDefined;
  expect(player.find('h3').text()).toBe('house');
  expect(player.find('button')).toBeUndefined;
});



