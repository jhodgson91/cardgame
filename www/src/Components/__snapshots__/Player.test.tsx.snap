// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same as last time with a button 1`] = `
<div
  className="grid-x grid-margin-x"
>
  <div
    className="small-2 small-offset-1"
  >
    <h3>
      house
    </h3>
  </div>
  <div
    className="small-2"
  >
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Play card
    </button>
  </div>
  <div
    className="cell player"
    id="house"
  >
    <div
      className="grid-container"
    >
      <div
        className="grid-x grid-padding-x"
      >
        <div
          id="test"
        >
          Text
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the same as last time without a button 1`] = `
<div
  className="grid-x grid-margin-x"
>
  <div
    className="small-2 small-offset-1"
  >
    <h3>
      house
    </h3>
  </div>
  <div
    className="cell player"
    id="house"
  >
    <div
      className="grid-container"
    >
      <div
        className="grid-x grid-padding-x"
      >
        <div
          id="test"
        >
          Text
        </div>
      </div>
    </div>
  </div>
</div>
`;
